### PR TITLE
backblaze-b2: skip broken tests causing build failure

### LINKED ISSUE #298601: Build failure: backblaze-b2
### Steps To Reproduce

1. `nix build github:nixos/nixpkgs/11d9b9e97bcb5091c8414bbf9e50eb0d5cd745a9#backblaze-b2`

`11d9b9e97bcb5091c8414bbf9e50eb0d5cd745a9` is the commit of current master.

### Build log

https://gist.github.com/linyinfeng/6f6f331ab82566dfd3f177b6438f05da

### Notify maintainers

@hrdinka @tomhoule

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.8.1, NixOS, 24.05 (Uakari), 24.05.20240319.b06025f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/etc/nix/inputs/nixpkgs`
```

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by alyraffauf**
## Description of changes

Fixes #298601. Disabled broken "TestConsoleTool", "TestNonUTF8TerminalSupport", and "TestVersionConsoleTool" tests causing build failure.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by tomhoule — APPROVED** [review]
Looks good, it builds and seems to work fine. Thanks!

**Review by starsep — APPROVED** [review]

**Review by hrdinka — APPROVED** [review]

File: pkgs/development/tools/backblaze-b2/default.nix [hunk 1]
```diff
@@ -60,6 +60,10 @@ python3Packages.buildPythonApplication rec {
 
     # fixed by https://github.com/Backblaze/B2_Command_Line_Tool/pull/915
     "TestRmConsoleTool"
+
+    "TestConsoleTool"
+    "TestNonUTF8TerminalSupport"
+    "TestVersionConsoleTool"
```
Comment (natsukium) [inline pkgs/development/tools/backblaze-b2/default.nix:66]: These failures are probably due to `pytest 7 -> 8`.
Could you please try replacing `pytestCheckHook` with `(pytestCheckHook.override { pytest = pytest_7; })`?
Then, in general, when you disable a test, you need to write a reason.

Comment (alyraffauf) [reply pkgs/development/tools/backblaze-b2/default.nix]: This worked! I pushed a new commit to the PR with these changes (and removing the unnecessarily disabled tests). Let me know if I need to make a new PR or change anything else.

**Review by natsukium — APPROVED** [review]
LGTM, thanks.

**Pull request merged**